spatialos-sdk/src/worker/locator.rs

```diff
@@ -197,68 +196,54 @@ impl Future for DeploymentListFuture {
     type Error = String;
 
     fn poll(&mut self) -> Result<Async<<Self as Future>::Item>, <Self as Future>::Error> {
-        if self.was_consumed {
+        if self.consumed {
             return Err("DeploymentListFuture has already been consumed".to_owned());
         }
 
-        assert!(!self.future.is_null());
+        assert!(!self.internal.is_null());
         let mut data: Option<Result<Vec<Deployment>, String>> = None;
         unsafe {
             Worker_DeploymentListFuture_Get(
-                self.future,
+                self.internal,
                 &0,
                 &mut data as *mut _ as *mut ::std::os::raw::c_void,
                 Some(DeploymentListFuture::callback_handler),
             );
         }
 
-        match data {
-            Some(result) => {
-                self.was_consumed = true;
-                match result {
-                    Ok(deployments) => Ok(Async::Ready(deployments)),
-                    Err(e) => Err(e),
-                }
-            }
-            None => Ok(Async::NotReady),
-        }
+        data.map_or(Ok(Async::NotReady), |result| {
+            self.consumed = true;
+            result.map(Async::Ready)
+        })
     }
 
     fn wait(self) -> Result<<Self as Future>::Item, <Self as Future>::Error>
     where
         Self: Sized,
     {
-        if self.was_consumed {
+        if self.consumed {
             return Err("DeploymentListFuture has already been consumed".to_owned());
         }
 
-        assert!(!self.future.is_null());
+        assert!(!self.internal.is_null());
         let mut data: Option<Result<Vec<Deployment>, String>> = None;
         unsafe {
             Worker_DeploymentListFuture_Get(
-                self.future,
+                self.internal,
                 ::std::ptr::null(),
                 &mut data as *mut _ as *mut ::std::os::raw::c_void,
                 Some(DeploymentListFuture::callback_handler),
             );
         }
 
-        match data {
-            Some(result) => match result {
-                Ok(deployments) => Ok(deployments),
-                Err(e) => Err(e),
-            },
-            None => {
-                panic!("Blocking call to Worker_DeploymentListFuture_Get did not trigger callback")
-            }
-        }
+        data.expect("Blocking call to Worker_DeploymentListFuture_Get did not trigger callback")
     }
 }
 
 impl Drop for DeploymentListFuture {
     fn drop(&mut self) {
-        if !self.future.is_null() {
-            unsafe { Worker_DeploymentListFuture_Destroy(self.future) }
+        if !self.internal.is_null() {
```

Comment:
Same question here: Does it make sense to be doing the null check in `drop`?

Reply:
Yup - moved to an `assert` in `new()` instead